src/Ledger/Foreign/HSLedger.agda

```diff
@@ -93,50 +93,57 @@ HSScriptStructure = record { p1s = HSP1ScriptStructure ; ps = HSP2ScriptStructur
 
 open import Data.Nat
 open import Data.Nat.Properties using (+-0-commutativeMonoid; _≟_)
-open import Ledger.TokenAlgebra
+open import Ledger.TokenAlgebra.Base using (TokenAlgebra)
 open import Ledger.Transaction
 
-coinTokenAlgebra : TokenAlgebra
-coinTokenAlgebra = record
-  { PolicyId                = ℕ
-  ; Value-CommutativeMonoid = +-0-commutativeMonoid
-  ; coin                    = id
-  ; inject                  = id
-  ; policies                = λ x → ∅
-  ; size                    = λ x → 1 -- there is only ada in this token algebra
-  ; property                = λ x → refl
-  ; coinIsMonoidMorphism    = record
-    { mn-homo = record
-      { sm-homo = record { ⟦⟧-cong = λ z → z ; ∙-homo  = λ x y → refl }
-      ; ε-homo  = refl
-      }
-    }
-  ; _≤ᵗ_                    = _≤_
-  ; DecEq-Value             = record { _≟_ = Data.Nat._≟_ }
-  }
+
+module _ where
+  open TokenAlgebra
+  open IsCommutativeMonoidMorphism
+  coinTokenAlgebra : TokenAlgebra
+  coinTokenAlgebra .Value-CommutativeMonoid  = +-0-commutativeMonoid
```

Comment:
make the style consistent with the definition below of `HSTransactionStructure`.

(By the way, why is it okay to use "Structure" as a suffix, but not "Type"?)

Reply:
Good point. I don't like the 'Structure' suffix either, if you have a better candidate let me know!